system/lib/fetch/wasmfs.cpp

```diff
@@ -0,0 +1,55 @@
+// // Copyright 2021 The Emscripten Authors.  All rights reserved.
+// // Emscripten is available under two separate licenses, the MIT license and the
+// // University of Illinois/NCSA Open Source License.  Both these licenses can be
+// // found in the LICENSE file.
+
+#include <stdlib.h>
+#define __NEED_struct_iovec
+#include <emscripten/emscripten.h>
+#include <wasi/api.h>
+
+extern "C" {
+
+  __wasi_errno_t __wasi_fd_write( __wasi_fd_t fd, const __wasi_ciovec_t *iovs, size_t iovs_len, __wasi_size_t *nwritten) {
+    if (fd == 1 || fd == 2) {
```

Comment:
This hardcoding of file descriptors 1 and 2 is temporary, right? It would be good to add a comment saying what this code does and that it is temporary behavior just to get something running at all.

Reply:
I think it should be a permanent convention. We don't want to tie printing to require to pull in the filesystem. Making the filesystem somehow define stdio file handles would certainly create such a dependency.

Reply:
Yes, I agree that's an important goal here.

I'm hoping that we can find a way to make that happen even without hardcoding stuff here, just relying on LLVM. But if we don't, let's hardcode.

As background, currently we avoid linking in any FS code when only stdout/stderr are used here:

https://github.com/emscripten-core/emscripten/blob/aca7c128fd844d2b2b306aa2d5205a07f053bbe0/emscripten.py#L65

For WASMFS we could perhaps do something similar. I was hoping we'd find a less hackish mechanism, perhaps using weak linking, but I think all options should be on the table.